Percy/Percy.cs

```diff
@@ -136,18 +175,143 @@ private static string GetPercyDOM()
                 else
                 {
                     data.TryGetProperty("type", out JsonElement type);
+                    data.TryGetProperty("widths", out JsonElement widths);
+                    data.TryGetProperty("config", out JsonElement config);
+                    setEligibleWidths(widths);
                     setSessionType(type.ToString());
+                    setCliConfig(config);
                     return (bool) (_enabled = true);
                 }
             }
             catch (Exception error)
             {
                 Log("Percy is not running, disabling snapshots");
-                if (DEBUG) Log<Exception>(error);
+                Log<Exception>(error, "debug");
                 return (bool) (_enabled = false);
             }
         };
 
+        private static dynamic getSerializedDom(WebDriver driver, object cookies, string opts) {
+            string script = $"return PercyDOM.serialize({opts})";
+            var domSnapshot = (Dictionary<string, object>)driver.ExecuteScript(script);
+            domSnapshot["cookies"] = cookies;
+            return domSnapshot;
+        }
+
+        private static List<int> GetWidthsForMultiDom(int[] widths)
+        {
+            var fetchedWidthsElement = (JsonElement)eligibleWidths;
+            var allWidths = fetchedWidthsElement.GetProperty("mobile")
+                                            .EnumerateArray()
+                                            .Select(x => x.GetInt32())
+                                            .ToList();
+
+            if (widths.Length != 0)
+            {
+                allWidths.AddRange(widths);
+            }
+            else
+            {
+                allWidths.AddRange(fetchedWidthsElement.GetProperty("config")
+                                                .EnumerateArray()
+                                                .Select(x => x.GetInt32()));
+            }
+
+            return allWidths.Distinct().ToList();
+        }
+
+        // Method to check if ChromeDriver supports CDP by checking the existence of ExecuteCdpCommand
+        private static bool IsCdpSupported(ChromeDriver chromeDriver)
+        {
+            return chromeDriver.GetType().GetMethod("ExecuteCdpCommand") != null;
+        }
+
+        private static void ChangeWindowDimensionAndWait(WebDriver driver, int width, int height, int resizeCount)
+        {
+            try
+            {
+                // Check if the driver is ChromeDriver and supports CDP
+                if (driver is ChromeDriver chromeDriver && IsCdpSupported(chromeDriver))
```

Comment:
This basically means that if the driver is RemoteWebDriver this will never work.

Reply:
Tested it with chrome driver in local and it is working. we can connect i can show you